python/sglang/srt/models/deepseek_v2.py

```diff
@@ -361,6 +348,25 @@ def __init__(
             ),
         )
 
+        apply_routed_scaling_factor_on_output = isinstance(
+            self.experts, FusedMoE
+        ) and isinstance(self.experts.quant_method, ModelOptNvFp4FusedMoEMethod)
```

Comment:
Curious if this needs to happen in the model file since it is specific to ModelOptNvFp4FusedMoEMethod. Can this setting be passed in with select_experts so that it can be reused for other moe models too?

Reply:
Pushed a change, could you please take a look?

Reply:
yeah, this looks good to me.